helpers/readingTime.js

```diff
@@ -0,0 +1,28 @@
+// constants for calculating reading time of an article.
+
+const wordsPerMinute = 275;
+
+/**
+   * input any string.
+   * output 'keeps all word characters
+   * returns the total number of characters'
+   */
+
+const characterCount = (body) => {
+  const reg = new RegExp(/\w/, 'g');
+  return (body.match(reg) || []).length;
+};
+
+
+const totalReadTime = (body, wordsPerMin = wordsPerMinute) => {
+  const wordcount = characterCount(body);
+  const wordtime = wordcount / wordsPerMin;
```

Comment:
@aaronsekisambu On the top of this file, you define this variable `wordsPerMinute` which I think is the number of words per minute not characters per minute. Then in the `characterCount` function the regex you are using counts the characters in the body, not words. This is giving rather an erroneous total reading time. Plus if you add an image nothing changes, unless we are not including them.  The first picture was a response I got with this regex used `const reg = new RegExp(/\w/, 'g');`. The second one is after changing it to counting words by grouping `const reg = new RegExp(/(\w{1,})/, 'g')
`  You see that according to the size of the body we have, the second one makes sense.
<img width="1150" alt="counting_characters" src="https://user-images.githubusercontent.com/24508902/57010343-6dd33680-6bfc-11e9-8502-ab70bd83c1a2.png">

<img width="1129" alt="counting_words" src="https://user-images.githubusercontent.com/24508902/57010405-bc80d080-6bfc-11e9-8ae2-4787836acb3a.png">

Reply:
@avpaul Thank you so much for the feedback given.
However, for including images counter, during our sprint retro your well aware that we agreed to ignore them at this stage. At this point in time, it's to not very useful to calculate the time spend on images which are about 12secs going down each.

For calculate words by grouping, this was my initial algorithm but a concern was raised. if the words are long and it takes an adult more time to read them, how would do you cater for that one. In that senario, I decided to get the wordCount per min and use it to count characters.

However, let value the two ideas and see which one is more efficient.

Reply:
@aaronsekisambu For images I get it. But for the other part, your concerns are valid. The problem is implementation. **You are comparing the number of words someone can read per minute to the total number of characters in the body!** It's words per minute not characters per minute. If you want to use characters then the variable you are using `wordsPerMin` should be characters per minute. You are using the correct value but you need to convert it in the correct measurement.  Plus the value([defined here](https://github.com/andela/ah-kgl-avengers-backend/blob/39ff67b64410ef52b1b5f48b7bef8ae2dcbdb26e/helpers/readingTime.js#L3)) **275 words per minute** is somehow large, from [this Google search](https://www.google.com/search?q=how+many+words+can+i+read+per+minute&oq=how+many+words+can+i+read+per+minute&aqs=chrome..69i57j0l5.10692j0j1&sourceid=chrome&ie=UTF-8) many results estimates it in [150 -200] and few others [200 -250]

Reply:
@avpaul thank you so much. I made more research and found out that it could be around [250](https://www.google.com/search?ei=V-DJXOXuIoicac7NltgI&q=avearage+adult+read+time&oq=avearage+adult+read+time&gs_l=psy-ab.3...46844.55267..55956...0.0..0.463.6021.2-21j2j1......0....1..gws-wiz.......0i71j0i131j0j0i131i67j0i3j0i10j0i67j0i13j0i22i30.z6f2Gm7zohs). I think I will slow it down to 230 to cater for all age groups and compensate for the images in that case

Reply:
@avpaul thank you for the fix in the grouping, I have tried it and I think it's viable.

Reply:
@avpaul here on this `const reg = new RegExp(/(\w{1,})/, 'g')` I think this `const reg = new RegExp(/(\w{1})+/, 'g')` is more readable and efficient in terms of performace. 
What do you think?

Reply:
That one is more efficient and clear. Just change the naming in this file so that it can be more clear for someone else reading the code. i.e: the function on line 11 is named **characterCount** better make it clear like **countWords**